XenonUtils/xex_patcher.cpp

```diff
@@ -403,7 +403,71 @@ XexPatcher::Result XexPatcher::apply(const uint8_t* xexBytes, size_t xexBytesSiz
             memmove(outDataCursor, srcDataCursor, blocks[i].dataSize);
         }
     }
-    else if (fileFormatInfo->compressionType == XEX_COMPRESSION_NORMAL || fileFormatInfo->compressionType == XEX_COMPRESSION_DELTA)
+    else if (fileFormatInfo->compressionType == XEX_COMPRESSION_NORMAL)
+    {
+        const Xex2CompressedBlockInfo* blocks = &((const Xex2FileNormalCompressionInfo*)(fileFormatInfo + 1))->firstBlock;
+        const uint32_t exeLength = xexBytesSize - xexHeader->headerSize.get();
+        const uint8_t* exeBuffer = &outBytes[headerTargetSize];
+
+        uint8_t* compressBuffer = NULL;
+        const uint8_t* p = NULL;
+        uint8_t* d = NULL;
+        sha1::SHA1 s;
+
+        compressBuffer = (uint8_t*)calloc(1, exeLength);
```

Comment:
You can use `std::unique_ptr<uint8_t[]>/std::make_unique<uint8_t[]>` for the allocation instead of allocating it manually.

Reply:
I've not used before std::make_unique<uint8_t[]> so this is new to me but I think I have used it correctly from a quick read about it